specs/Freeze.md

```diff
@@ -0,0 +1,155 @@
+latest data and the page ready to show fresh content when it becomes visible.
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+```IDL
+interface ICoreWebView2_2 : ICoreWebView2 {
+
+  /// An app may call the `TryFeezeAsync` API to have the WebView2 consume less memory.
+  /// This is useful when a Win32 app becomes invisible, or when a Universal Windows
+  /// Platform app is suspended, during the suspended event handler before completing
+  /// the suspended event.
+  /// The CoreWebView2Controller must be invisible when the API is called. Otherwise, the
+  /// API fails with `HRESULT_FROM_WIN32(ERROR_INVALID_STATE)`.
+  /// Freezing is similar to putting a tab to sleep in the Edge browser. Freezing pauses
+  /// WebView script timers and animations, minimizes CPU usage for the associated
+  /// browser renderer process and allows the operating system to reuse the memory that was
+  /// used by the renderer process for other processes.
+  /// Note that the freeze is best effort and considered completed successfully once the request
+  /// is sent to browser renderer process. If there is a running script, the script will continue
+  /// to run and the renderer process will be frozen after that script is done.
+  /// See [Sleeping Tabs FAQ](https://techcommunity.microsoft.com/t5/articles/sleeping-tabs-faq/m-p/1705434)
+  /// for conditions that might prevent WebView from being frozen. In those situations,
+  /// The TryFreeze operation will fail and the completed handler will be invoked with isSuccessful as false.
+  /// The WebView will be automatically unfrozen when it becomes visible. Therefore, the
+  /// app normally doesn't have to call Unfreeze.
+  /// The app can call `Unfreeze` and then `TryFreeze` periodically for an invisibile WebView so that
+  /// the invisible WebView can sync up with latest data and the page ready to show fresh content
+  /// when it becomes visible.
+  HRESULT TryFreeze([in] ICoreWebView2StagingTryFreezeCompletedHandler* handler);
+
+  /// Unfreeze the WebView so that it would resume activities on the web page.
+  /// This API can be called while the WebView2 controller is invisible.
+  /// The app can interact with the WebView immediately after unfreeze.
+  /// WebView will be automatically unfrozen when it becomes visible.
+  HRESULT Unfreeze();
+}
+
+/// The caller implements this interface to receive the TryFreeze result.
+[uuid(00F206A7-9D17-4605-91F6-4E8E4DE192E3), object, pointer_default(unique)]
+interface ICoreWebView2StagingTryFreezeCompletedHandler : IUnknown {
+
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL isSuccessful);
+
+}
+```
+## .Net WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    public partial class CoreWebView2
+    {
+        // There are other API in this interface that we are not showing 
+        public Task<bool> TryFreezeAsync();
+        public void Unfreeze();
```

Comment:
Question for reviewers:
would it better to have FreezeAsync/UnfreezeAsync so that we have matching async for the pair of the APIs and don't have a TryXXX method without XXX method?
Should we name it as Freeze/Unfreeze or Suspend/Resume?